pkg/util/bulk/tracing_aggregator.go

```diff
@@ -113,27 +102,16 @@ func TracingAggregatorEventToBytes(
 
 var _ tracing.EventListener = &TracingAggregator{}
 
-// MakeTracingAggregatorWithSpan returns an instance of an TracingAggregator along with a
-// newly created child context. The TracingAggregator is registered as a
-// tracing.EventListener on the span associated with newly created context.
-//
-// The TracingAggregator instance is responsible for finishing the returned span, and
-// so the user must call Close().
-func MakeTracingAggregatorWithSpan(
-	ctx context.Context, aggregatorName string, tracer *tracing.Tracer,
-) (context.Context, *TracingAggregator) {
+func MakeTracingAggregator(ctx context.Context) *TracingAggregator {
 	if tracing.SpanFromContext(ctx) == nil {
-		log.Warningf(ctx, "tracing aggregator %s cannot be created without a tracing span", aggregatorName)
-		return ctx, nil
+		log.Warning(ctx, "tracing aggregator cannot be created without a tracing span")
+		return nil
 	}
 	agg := &TracingAggregator{}
-	aggCtx, aggSpan := tracing.EnsureChildSpan(ctx, tracer, aggregatorName,
-		tracing.WithEventListeners(agg))
 
 	agg.mu.Lock()
 	defer agg.mu.Unlock()
 	agg.mu.aggregatedEvents = make(map[string]TracingAggregatorEvent)
-	agg.sp = aggSpan
```

Comment:
[nit] You could put the map creation directly in the original declaration of agg. Alternatively, since this is the first place you created agg and no one else possibly has a pointer to it, you don't need to lock it.

Reply:
done